src/cpp/cpp_constructor.cpp

```diff
@@ -47,12 +47,16 @@ codet cpp_typecheckt::cpp_constructor(
     // The purpose of the tag #array_ini is to rule out ill-formed
     // programs.
 
+    // TODO: in a program it is not allowed, but the implementation of the
```

Comment:
I don't think this comment is very clear ( @peterschrammel )

Reply:
De/activating this code has no effect on cpp and systemc regression tests. We should remove this change from this PR.

Reply:
I have removed the commit "Deactivate internal array initializer".